### ISSUE TITLE
Incorrect lifetime reported for `using` local

### ISSUE THREAD

**Issue by hamarb123** [dotnet/roslyn#85694]
**Version Used**: .NET SDK 10.0.401

**Steps to Reproduce**:

```csharp
using System;
using System.Diagnostics.CodeAnalysis;

public struct S : IDisposable
{
    [UnscopedRef]
    public ref S Self => ref this;

    public void Dispose() { }

    public S X() => new();
}

public class C
{
    public static void X1(ref S s)
    {
        using var a = s.X();
        using var b = s.X();

        ref var c = ref a.Self;
        if ("1" == 1.ToString()) c = ref b.Self; // No CS8374
    }

    public static void X2(ref S s)
    {
        using var a = s.X();
        using var b = s.X();

        ref var c = ref a.Self;
        if ("1" == 1.ToString())
        {
            c = ref b.Self; // CS8374
        }
    }

    public static void X3(ref S s)
    {
        using var a = s.X();
        using var b = s.X();

        ref var c = ref b.Self;
        if ("1" == 1.ToString())
        {
            c = ref a.Self; // CS8374
        }
    }
}
```

[Godbolt](https://godbolt.org/z/PrabGnb3K)

[.NET Lab](https://lab.razor.fyi/#zZBBSsNAGIVRpOisxBM8uko2gerOEEHaja7EKBRFcDKdpANpJuSfVkrpNTxIr-ABvEH33kImadOmCoIL8V_NzPvf433Dli3GbgqdFHzkCTp5b41JZQnCKRk58tn2zespnmSajBLkdfVAXmY8nZIin7F8HKVKgEwxFgYhznHVU5Rr4lEq2YwBwON9RkLncnAr46fyZeUqZIwQoUxjBBflzQzL1K2diVYDVJHScTHDvCGH6DuudWfyxXF9Nq8riZQTobvqUPfkZh3a7zhVAXLLlWrRTgU_4QU4ApDXt8lfxWgj1qpNtJpAUJ65Z_E2bhXDaXfaCAJ0vDsdmkJlieO6tSHaMjRRG91P_1H3emHTws53RCipfmA7-wO23VK_ZNv9ojXbnF3vC3o4Oly8fSxeh8cHz3uf)

**Diagnostic Id**:

CS8374: `Cannot ref-assign 'b.Self' to 'c' because 'b.Self' has a narrower escape scope than 'c'.`
CS8374: `Cannot ref-assign 'a.Self' to 'c' because 'a.Self' has a narrower escape scope than 'c'.`

**Expected Behavior**:

Compiles without error - this would match what the lifetimes are. Or potentially, only one of the `X2` and `X3` ordering works if you consider `a` and `b` to have slightly different lifetime scopes due to `using` effectively translating to a `try/finally` - although, I am not sure that this can actually ever make a difference here (ito blocking something actually unsafe) (I suspect not, but haven't thought about it super hard) (regardless, even if this were the case, then `X3` should still compile as can be seen in https://godbolt.org/z/bPYsd53Te, and it certainly shouldn't depend on whether I put the assignment into a block or inline statement).

**Actual Behavior**:

`X1` compiles fine, but `X2` and `X3` have errors. If you change them from `using` locals to normal locals, or if you change it to `ref readonly var c = ...` and from `ref S Self` to `readonly ref readonly S Self`, then it compiles.

/cc @jaredpar @jjonescz
